### PR TITLE
extended windows support for EN characters

### PR DESCRIPTION
**Pull request by c-v-ya**
Typo check in example.todo.
Added windows and linux text editors. Reworked encoding but it crashes if sees non-english character in line. Works fine if todo file in english.
Spaces replaced with tabs (sorry :sweat_smile:).

### CODE REVIEW

**Comment by eugene-katsevman** [comment]
Kindly use spaces, the diff is completely unreadable otherwise

**Comment by c-v-ya** [comment]
I guess commits doesn't ring a notification... Or u r just busy, but anyway. I pushed a new commit, replaced tabs with spaces.

File: keeper.py [hunk 1]
```diff
@@ -14,6 +14,11 @@
 
 WINDOWS = platform.system() == "Windows"
 
+if WINDOWS:
+    import sys
+    reload(sys)
```
Comment (eugene-katsevman) [inline keeper.py]: why?

File: keeper.py [hunk 2]
```diff
@@ -118,8 +123,14 @@ def random_tasks(arg):
 
 
 def edit(arg):
-    if args.filenames:
-        os.system("gedit "+" ".join([tasks.get_dir()+fn + ".todo" for fn in args.filenames]))
+    editors_win = ['notepad']
+    editors_unix = ['gedit', 'kwrite', 'pluma', 'leafpad']
+    if WINDOWS:
```
Comment (eugene-katsevman) [inline keeper.py]: and what if args,filenames is empty?

File: keeper.py [hunk 3]
```diff
@@ -118,8 +123,14 @@ def random_tasks(arg):
...
+        for editor in editors_win:
+            os.system(editor+" "+" ".join([tasks.get_dir()+fn + ".todo" for fn in args.filenames]))
```
Comment (eugene-katsevman) [inline keeper.py]: starting every single editor with the same file opened?

File: keeper.py [hunk 4]
```diff
@@ -118,8 +123,14 @@ def random_tasks(arg):
...
+    elif args.filenames:
+        for editor in editors_unix:
+            os.system(editor+" "+" ".join([tasks.get_dir()+fn + ".todo" for fn in args.filenames]))
```
Comment (eugene-katsevman) [inline keeper.py]: same here, we should stop on first successful os.system call

File: tasks.py [hunk 5]
```diff
@@ -19,6 +19,8 @@ def set_line(filename, lineno, line):
 
 
 def get_dir():
```
Comment (eugene-katsevman) [inline tasks.py:21]: better rewrite it with os.sytem.join

**Comment by eugene-katsevman** [comment]
finished the review, just two or three simple fixes to go

**Comment by c-v-ya** [comment]
So, it chooses default system text editor now or asking to set it up (for linux users). `get_dir` function rewritten with `os.getcwd()`. I didn't really understand why `os.system` and `join`.

And what if args,filenames is empty? Well, if u mean what happens when `keeper edit`, then there is an error `sh: todo: command not found` obviously. Need to implement foolproof with some message like "uh-uh, specify the filename". Or to create default file with some info message.

**Pull request closed without merging**